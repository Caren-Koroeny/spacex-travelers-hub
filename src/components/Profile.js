import React from "react";

const Profile = () => {
  return(
    <div className="container">
    <div className="profileContainer">
      <div className="myMissions">
        <h2>My Missions</h2>
        <div className="cardsContainer">{}</div>
      </div>
      <div className="myRockets">
        <h2>My Rockets</h2>
        <div className="cardsContainer">{}</div>
      </div>
    </div>
  </div>
  );
}

export default Profile;
