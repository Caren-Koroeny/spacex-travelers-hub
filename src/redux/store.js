import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketReducer from './rockets/rocketSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,

    rockets: rocketReducer,
  },
});

export default store;
