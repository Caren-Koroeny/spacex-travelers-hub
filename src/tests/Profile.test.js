import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../redux/store';
import Profile from '../components/Profile';

describe('Display profile components', () => {
  it('Should display Profile items', () => {
    const container = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
