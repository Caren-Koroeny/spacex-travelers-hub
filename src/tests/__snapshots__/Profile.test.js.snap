// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Mission Components Should display Mission items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="missionContainer"
      >
        <div
          class="headers"
        >
          <div>
            <h2>
              Mission
            </h2>
          </div>
          <div>
            <h2>
              Description
            </h2>
          </div>
          <div>
            <h2>
              Status
            </h2>
          </div>
          <div />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="missionContainer"
    >
      <div
        class="headers"
      >
        <div>
          <h2>
            Mission
          </h2>
        </div>
        <div>
          <h2>
            Description
          </h2>
        </div>
        <div>
          <h2>
            Status
          </h2>
        </div>
        <div />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display Profile Components Should display Mission items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="myMissions"
          >
            <h2>
              My Missions
            </h2>
            <div
              class="cardsContainer"
            >
              <p>
                No missions booked yet
              </p>
            </div>
          </div>
          <div
            class="myRocket"
          >
            <h2>
              My Rockets
            </h2>
            <div
              class="cardsContainer"
            >
              <p>
                No rockets have been reserved
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="myMissions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="cardsContainer"
          >
            <p>
              No missions booked yet
            </p>
          </div>
        </div>
        <div
          class="myRocket"
        >
          <h2>
            My Rockets
          </h2>
          <div
            class="cardsContainer"
          >
            <p>
              No rockets have been reserved
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Display Profile Components Should display Profile items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="myMissions"
          >
            <h2>
              My Missions
            </h2>
            <div
              class="cardsContainer"
            >
              <p>
                No missions booked yet
              </p>
            </div>
          </div>
          <div
            class="myRocket"
          >
            <h2>
              My Rockets
            </h2>
            <div
              class="cardsContainer"
            >
              <p>
                No rockets have been reserved
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="myMissions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="cardsContainer"
          >
            <p>
              No missions booked yet
            </p>
          </div>
        </div>
        <div
          class="myRocket"
        >
          <h2>
            My Rockets
          </h2>
          <div
            class="cardsContainer"
          >
            <p>
              No rockets have been reserved
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navigation component should render Header component 1`] = `
<div>
  <header
    class="headerSection"
  >
    <section
      class="logoSection"
    >
      <img
        alt="logo"
        class="logo"
        src="planet1.png"
      />
      <h1>
        Space Travelers Hub
      </h1>
    </section>
    <nav
      class="navLinks"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
        style="color: rgb(100, 149, 237); font-weight: 700; text-decoration: underline;"
      >
        Rockets
      </a>
      <a
        class=""
        href="/missions"
        style="font-weight: regular;"
      >
        Missions
      </a>
      <div>
        |
      </div>
      <a
        class=""
        href="/myprofile"
        style="font-weight: regular;"
      >
        My Profile
      </a>
    </nav>
  </header>
  <hr />
</div>
`;
